### PR TITLE
[EN DateTimeV2] Support for ranges with "nights" (#1789)

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
Fix to issue #1789.
Added relevant test cases in DateTimeModel.
The result of the existing test case "Can I do a booking for the 09th of May for 2 nights?" has been modified to match the one from "Can I do a booking for the 09th of May for 2 days?"

### CODE REVIEW

File: Patterns/English/English-DateTime.yaml [hunk 1]
```diff
@@ -346,7 +346,7 @@ IshRegex: !nestedRegex
   def: '\b({BaseDateTime.HourRegex}(-|——)?ish|noon(ish)?)\b'
   references: [ BaseDateTime.HourRegex ]
 TimeUnitRegex: !simpleRegex
-  def: ([^A-Za-z]{1,}|\b)(?<unit>h(ou)?rs?|h|min(ute)?s?|sec(ond)?s?)\b
+  def: ([^A-Za-z]{1,}|\b)(?<unit>h(ou)?rs?|h|min(ute)?s?|sec(ond)?s?|nights?)\b
```
Comment (tellarin) [inline Patterns/English/English-DateTime.yaml:349]: Why is this change necessary? "nights" don't specify time.

File: Patterns/English/English-DateTime.yaml [hunk 2]
```diff
@@ -477,7 +477,7 @@ LessThanRegex: !simpleRegex
 MoreThanRegex: !simpleRegex
   def: \b(more\s+than)\b
 DurationUnitRegex: !nestedRegex
-  def: (?<unit>{DateUnitRegex}|h(ou)?rs?|h|min(ute)?s?|sec(ond)?s?)\b
+  def: (?<unit>{DateUnitRegex}|h(ou)?rs?|h|min(ute)?s?|sec(ond)?s?|nights?)\b
```
Comment (tellarin) [inline Patterns/English/English-DateTime.yaml:480]: Why not make this change in DateUnitRegex? It's conceptually right and where similar cases for "days" are captured.

Comment (LionbridgeCSII) [reply Patterns/English/English-DateTime.yaml:480]: Because then expressions like "last night" will be resolved as "daterange" instead of "datetimerange"

File: Patterns/English/English-DateTime.yaml (hunk 1 above)
Comment (LionbridgeCSII) [reply Patterns/English/English-DateTime.yaml:349]: yes right, it is not necessary. Removed

File: Patterns/English/English-DateTime.yaml (hunk 2 above)
Comment (tellarin) [reply Patterns/English/English-DateTime.yaml:480]: Seems reasonable. But ideally the term should then be added closes to {DateUnitRegex} in the regex. Not a merge blocker though.
Thanks.
